nn_sdwan/datadog_checks/nn_sdwan/check.py

```diff
@@ -0,0 +1,135 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+# import net
+
+import platform
+import subprocess
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .net_api import vManageApi
+
+
+def pingable(host: str):
+    """
+    Returns True if host (str) responds to a ping request.
+    """
+
+    param = '-n' if platform.system().lower() == 'windows' else '-c'
+    command = ['ping', param, '1', host]
+    return subprocess.call(command, stdout=subprocess.DEVNULL) == 0
+
+
+class NnSdwanCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'nn_sdwan'
+
+    def __init__(self, name, init_config, instances):
+        super(NnSdwanCheck, self).__init__(name, init_config, instances)
+
+        # Validate instance config fields are present
+        for field in 'hostname username password protocol'.split():
+            if not self.instance.get(field):
+                raise ConfigurationError(f'Configuration error: "{field}" not present.')
+
+        self.vmanage_api = vManageApi(instance=instances[0])
+
+    def check(self, _):
+        # Handle service check (availablity / reachability)
+        if not pingable(self.instance.get('hostname')):
+            self.service_check('nn_sdwan.sdwan_controller.online', self.CRITICAL, message='Host unreachable.')
+            return
+
+        try:
+            self.vmanage_api.get_device_info()
+            # vManage is online and responding
+            self.service_check('nn_sdwan.sdwan_controller.online', self.OK)
+        except Exception:
+            self.service_check('nn_sdwan.sdwan_controller.online', self.WARNING, message='vManage API is not responding.')
```

Comment:
Style tests are failing here: `./datadog_checks/nn_sdwan/check.py:50:121: E501 line too long (122 > 120 characters)`
Can you try running `ddev -e test nn_sdwan -fs`?